### PR TITLE
feat: hiding mobile number section for IN cluster users - part 1

### PR DESCRIPTION
**Pull request by Aniruddha-Shriwant**
Will hide the info task of text receipts in next PR

## Clickup
https://app.clickup.com/t/86cvedzha

## Code Coverage
Please add code coverage here

## UI Preview
For users except IN_CLUSTER
image

For IN_CLUSTER
image

### CODE REVIEW

File: src/app/core/services/tasks.service.ts [hunk 1]
```diff
@@ -430,10 +432,16 @@ export class TasksService {
     return forkJoin({
       rtfEnrolledCards: rtfEnrolledCards$,
       eou: from(this.authService.getEou()),
+      isUserFromINCluster: from(this.utilityService.isUserFromINCluster()),
     }).pipe(
-      switchMap(({ rtfEnrolledCards, eou }) => {
-        //Show this task only if mobile number is not verified and user is enrolled for RTF
-        if (!eou.ou.mobile_verified && eou.ou.mobile_verification_attempts_left !== 0 && rtfEnrolledCards.length) {
+      switchMap(({ rtfEnrolledCards, eou, isUserFromINCluster }) => {
```
Comment (Aniruddha-Shriwant) [inline src/app/core/services/tasks.service.ts:437]: TODO for part 2 pr - add test coverage for this

File: src/app/fyle/my-profile/employee-details-card/employee-details-card.component.html [hunk 2]
```diff
@@ -21,7 +24,7 @@
       </ion-row>
     </ion-grid>
   </div>
-  <div class="employee-details-card__bottom-section">
+  <div *ngIf="isMobileNumberSectionVisible" class="employee-details-card__bottom-section">
```
Comment (Aniruddha-Shriwant) [inline src/app/fyle/my-profile/employee-details-card/employee-details-card.component.html:27]: TODO for part 2 pr - add test coverage for this

File: src/app/core/services/utility.service.ts [hunk 3]
```diff
@@ -154,6 +156,10 @@ export class UtilityService {
     return result;
   }
 
+  isUserFromINCluster(): Promise<boolean> {
+    return this.tokenService.getClusterDomain().then((clusterDomain) => clusterDomain.includes('in1.fylehq.com'));
```
Comment (Chethan-Fyle) [inline src/app/core/services/utility.service.ts:160]: There are some American customers in India cluster. Are we hiding this feature for them too?

**Review by Chethan-Fyle — COMMENTED** [review]
One doubt.

**Review by Chethan-Fyle — APPROVED** [review]

File: src/app/core/services/tasks.service.ts [hunk 4]
```diff
@@ -430,10 +432,16 @@ export class TasksService {
...
+        //Show this task only if mobile number is not verified and user is enrolled for RTF and user is not from IN cluster
+        if (
+          !eou.ou.mobile_verified &&
```
Comment (suyashpatil78) [inline src/app/core/services/tasks.service.ts:440]: should we add USD/CAD condition here?

Comment (Aniruddha-Shriwant) [reply src/app/core/services/tasks.service.ts:440]: `rtfEnrolledCards` this is there so skipping 😛

**Review by suyashpatil78 — APPROVED** [review]

File: src/app/core/services/tasks.service.ts (hunk 4 above)
Comment (suyashpatil78) [reply src/app/core/services/tasks.service.ts:440]: got it

**Pull request merged**